#!/usr/bin/env python
PACKAGE = "velodyne_pointcloud"

from math import pi
import dynamic_reconfigure.parameter_generator_catkin as pgc

gen = pgc.ParameterGenerator()

gen.add("min_range", 
  pgc.double_t, 
  0, 
  "min range to publish", 
  0.9, 0.1, 10.0)
  
gen.add("max_range", 
  pgc.double_t, 
  0, 
  "max range to publish", 
  130, 0.1, 200)
  
gen.add("view_direction", 
  pgc.double_t, 
  0, 
  "angle defining the center of view",
  0.0, -pi, pi)
  
gen.add("view_width", 
  pgc.double_t, 
  0, 
  "angle defining the view width",
  2*pi, 0.0, 2*pi)
  
gen.add("frame_id", 
  pgc.str_t, 
  0, 
  "new frame of reference for point clouds", 
  "odom")

gen.add("organize_cloud",
  pgc.bool_t,
  0,
  "organize cloud",
  False)

exit(gen.generate(PACKAGE, "transform_node", "TransformNode"))
